worker/src/ee/cloudSpendAlerts/handleCloudSpendAlertJob.ts

```diff
@@ -0,0 +1,258 @@
+import { sendCloudSpendAlertEmail } from "@langfuse/shared/src/server";
+
+export const handleCloudSpendAlertJob = async (job: Job<{ orgId: string }>) => {
+  const { orgId } = job.data;
+
+  logger.info(`[CLOUD SPEND ALERTS] Processing org ${orgId}`);
+
+  if (!env.STRIPE_SECRET_KEY) {
+    logger.warn("[CLOUD SPEND ALERTS] Stripe secret key not found");
+    throw new Error("Stripe secret key not found");
+  }
+
+  // Fetch organization with spend alerts
+  const orgData = await prisma.organization.findFirst({
+    where: {
+      id: orgId,
+    },
+    include: {
+      cloudSpendAlerts: true,
+    },
+  });
+
+  if (!orgData) {
+    logger.error(`[CLOUD SPEND ALERTS] Organization ${orgId} not found`);
+    return;
+  }
+
+  const org = {
+    ...parseDbOrg(orgData),
+    spendAlerts: orgData.cloudSpendAlerts,
+  };
+
+  // Check if org has spend alerts configured
+  if (org.spendAlerts.length === 0) {
+    logger.info(`[CLOUD SPEND ALERTS] No spend alerts for org ${orgId}`);
+    return;
+  }
+
+  if (org.cloudConfig?.plan === "Hobby") {
+    // handle case where user has downgraded to hobby
+    logger.info(
+      `[CLOUD SPEND ALERTS] Org ${orgId} not entitled to spend alerts (plan: Hobby"})`,
+    );
+    return;
+  }
+
+  // Get Stripe customer ID
+  const stripeCustomerId = org.cloudConfig?.stripe?.customerId;
+  if (!stripeCustomerId) {
+    logger.error(
+      `[CLOUD SPEND ALERTS] Stripe customer id not found for org ${orgId}`,
+    );
+    traceException(
+      `[CLOUD SPEND ALERTS] Stripe customer id not found for org ${orgId}`,
+    );
+    return;
+  }
+  // Get Stripe subscription ID
+  const stripeSubscriptionId = org.cloudConfig?.stripe?.activeSubscriptionId;
+  if (!stripeSubscriptionId) {
+    logger.error(
+      `[CLOUD SPEND ALERTS] Stripe subscription id not found for org ${orgId}`,
+    );
+    traceException(
+      `[CLOUD SPEND ALERTS] Stripe subscription id not found for org ${orgId}`,
+    );
+    return;
+  }
+
+  // Setup stripe client
+  const stripe = new Stripe(env.STRIPE_SECRET_KEY);
+
+  try {
+    // Get subscription to check billing cycle
+    const subscription =
+      await stripe.subscriptions.retrieve(stripeSubscriptionId);
+
+    const currentPeriodStart = new Date(
+      subscription.current_period_start * 1000,
+    );
+
+    // Create preview invoice to calculate current spend
+    const canCreateInvoicePreview = [
+      "active",
+      "past_due",
+      "trialing",
+      "unpaid",
+    ].includes(subscription.status);
+
+    const previewInvoice = canCreateInvoicePreview
+      ? await backOff(
+          async () =>
+            await stripe.invoices.createPreview({
+              customer:
+                typeof subscription.customer === "string"
+                  ? subscription.customer
+                  : subscription.customer?.id,
+              subscription: stripeSubscriptionId,
+            }),
+          {
+            numOfAttempts: 3,
+          },
+        )
+      : null;
```

Comment:
I think we should "Exit" with a warning if the org cannot create preview invoices similar to the exit conditions above. This may produce "silent" exceptions otherwise.